### PR TITLE
Approach 2: Support for aarch64 - Using libseccomp FFI

### PR DESCRIPTION
**Pull request by chethanah**
Hello @KentaTada 

- libseccomp provides `seccomp_syscall_resolve_num_arch` API.
- Using `extern` foreign function interfaces (FFI) we can
- resolve the syscall num

### Cargo test

- **x86_64**
```bash
running 5 tests
test report::test_convert_syscall_id ... ok
test report::test_remove_raw_syscall ... ok
test report::test_generate_profile ... ok
test report::test_convert_error_info ... ok
test utils::test_search_tracefs_path ... FAILED

failures:

---- utils::test_search_tracefs_path stdout ----
thread 'utils::test_search_tracefs_path' panicked at 'assertion failed: std::fs::metadata(&tracefs_path.unwrap()).is_ok()', src/utils/mod.rs:21:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    utils::test_search_tracefs_path

test result: FAILED. 4 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--bin oci-ftrace-syscall-analyzer'
```

- **aarch64**
```bash
mendel@queen-wasp:~/oci-ftrace-syscall-analyzer$ uname -a 
Linux queen-wasp 4.14.98-imx #1 SMP PREEMPT Fri Nov 8 23:28:21 UTC 2019 aarch64 GNU/Linux

running 5 tests
test report::test_convert_syscall_id ... ok
test report::test_remove_raw_syscall ... ok
test report::test_generate_profile ... ok
test report::test_convert_error_info ... ok
test utils::test_search_tracefs_path ... FAILED

failures:

---- utils::test_search_tracefs_path stdout ----
thread 'utils::test_search_tracefs_path' panicked at 'assertion failed: std::fs::metadata(&tracefs_path.unwrap()).is_ok()', src/utils/mod.rs:21:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    utils::test_search_tracefs_path

test result: FAILED. 4 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--bin oci-ftrace-syscall-analyzer'
```
- Build on `aarch64` without errors
- Test for convert syscall id passes

### CODE REVIEW

**Comment by chethanah** [comment]
- Reason for test failure in above 
```
test utils::test_search_tracefs_path ... FAILED
```
- Since the above test case asserts the tracefs path, it requires sudo permissions
- For example
```
$ ls /sys/kernel/debug/tracing 
ls: cannot access '/sys/kernel/debug/tracing': Permission denied

// use sudo
$ sudo ls /sys/kernel/debug/tracing 
```
- Therefore run test again with sudo permissions
```
$ sudo /home/chethan/.cargo/bin/cargo test -- --nocapture
<skip warning>
    Finished test [unoptimized + debuginfo] target(s) in 0.01s
     Running target/debug/deps/oci_ftrace_syscall_analyzer-232073af4bba31db

running 5 tests
test report::test_remove_raw_syscall ... ok
test report::test_convert_error_info ... ok
test report::test_convert_syscall_id ... ok
test report::test_generate_profile ... ok
test utils::test_search_tracefs_path ... ok

test result: ok. 5 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/integ-fd0fcf68a6e305d1

running 11 tests
test integration::trace_does_not_start_with_annotations_false ... ok
test integration::trace_does_not_start_with_use_annots ... ok
test integration::report_conflict_between_livedump_and_profile ... ok
test integration::error_filter ... ok
test integration::set_buffer_size ... ok
test integration::trace_start ... ok
test integration::syscall_filter ... ok
test integration::report ... ok
test integration::report_livedump ... ok
test integration::report_profile ... ok
test integration::trace_start_with_annotations ... ok

test result: ok. 11 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out


```

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,4 @@
 target/
 Cargo.lock
 **/*.rs.bk
+.idea
```
Comment (KentaTada) [inline .gitignore:4]: Not to add unrelated changes.
Could you create the new pull request to add ".idea" to .gitignore?

File: src/report/mod.rs [hunk 2]
```diff
@@ -14,21 +13,54 @@ use std::fs::File;
...
-fn convert_syscall_id(line: &str, syscall_info: &HashMap<i64, &str>) -> Option<String> {
+#[link(name = "seccomp")]
+extern "C" {
+    fn seccomp_syscall_resolve_num_arch(
+        arch: libc::int32_t,
+        syscall: libc::c_int,
+    ) -> *const libc::c_char;
+}
+
+#[cfg(target_arch = "aarch64")]
```
Comment (KentaTada) [inline src/report/mod.rs:37]: I'm sorry but I don't know why each architecture has libseccomp_convert_syscall_id().
Is it needed?

**Comment by chethanah** [comment]
Hello @KentaTada 

- I've removed `.idea` from `.gitignore` which was added by IDE. This change is not required
- The reason for adding conditional build was to update the `arch_token` in `seccomp_syscall_resolve_num_arch` for aarch64 and x86_64
  - But verified when `arch_token=0` the native arch is taken
  - ref: [libseccomp api.c](https://github.com/seccomp/libseccomp/blob/master/src/api.c#L446-L447)
  - Therefore removed conditional build based on arch
- Updated the required changes, please verify 

**Basic Verification**
```bash
$ uname -m 
aarch64

$ cd bundle
$ cat rootfs/hello.sh 
#!/bin/bash 
cd /bin

# update hooks in config.json
$ jq .process.args config.json 
[
  "/hello.sh"
]
$ runc run test 
$ grep chdir ftrace_syscalls_dump.log 
        hello.sh-4568  [001] ...1 13538.476904: sys_chdir(filename: 40001de120)
        hello.sh-4568  [001] ...1 13538.476943: sys_chdir -> 0x0
        hello.sh-4568  [001] ...1 13538.477019: sys_fchdir(fd: 9)
        hello.sh-4568  [001] ...1 13538.477028: sys_fchdir -> 0x0
        hello.sh-4568  [001] ...1 13538.477222: sys_fchdir(fd: 7)
```

**Comment by KentaTada** [comment]
LGTM
